### DISCUSSION TITLE
Example 23 Second order integrator

### DISCUSSION THREAD

**Discussion by bslazarov** [mfem/mfem#4246] [Q&A]
Does anybody know the explicit form of the solution at step n+1 in example 23? I can see that d2udt2 = f(u + fac0*d2udt2,dudt + fac1*d2udt2, t); however, I am not sure why the solution at n+1, i.e., u+fac0*d2udt2, has contribution only from the acceleration term and does not include the velocity term.

**Comment by tzanio** [comment]
ping: @IdoAkkerman

**Comment by IdoAkkerman** [comment]
Short answer:
Both  the u and dudt variables in the Mult and ImplicitSolve should already be modified to include the know components of dudt and d2udt2.
This should be taken care of in the ODE solver step.

Longer answer:
What we solve for is,

f(u^a0, dudt^a1, d2udt2^a2) =0

Where a0, a1 and a2 are effectively timelevels (usually between n and n+1).
To arrive at a solvable system interpolation (extrapolation) rules relating u, dudt and d2udt2 and different timelevels need to be provided.
For a large part these defines the actual time integrator method used.


This can be rewritten as
f(c0 u^n +  c1 dudt^n + c2 d2udt2^n + fac0 d2udt2^a2, c2 dudt^n + c3 d2udt2^n + fac1 d2udt2^a2, c4 d2udt2^n + fac1 d2udt2^a2)

In the mult/implicitSolve routines we have the following association

d2udt2   —> d2udt2^a2 this the unknown acceleration (at the intermediate level) for which we solve(march)

Furthermore
u = c0 u^n +  c1 dudt^n + c2 d2udt2^n  
dudt = c0 u^n +  c1 dudt^n + c2 d2udt2^n   
These are the “known" parts of unknown u^a0 and dudt^a1.

**Comment by IdoAkkerman** [comment]
@4233 should fix the issue

**Reply by IdoAkkerman** [reply]
#4233
